test/deployframework/framework.go

```diff
@@ -196,3 +211,100 @@ func (df *DeployFramework) NewDeployerConfig(
 		MeteringConfig:           meteringConfig,
 	}, nil
 }
+
+// CreateRegistryResources is a deployframework method responsible
+// for instantiating a new CatalogSource that can be used
+// throughout individual Metering installations.
+func (df *DeployFramework) CreateRegistryResources(registryImage, meteringOperatorImage, reportingOperatorImage string) (string, string, error) {
+	deploymentManifestPath := filepath.Join(df.RepoDir, olmManifestsDir, registryDeploymentManifestName)
+	err := CreateRegistryDeployment(df.Logger, df.Client, deploymentManifestPath, registryImage, meteringOperatorImage, reportingOperatorImage, registryDeployNamespace)
+	if err != nil {
+		return "", "", fmt.Errorf("failed to create the registry deployment manifest in the %s namespace: %v", err, registryDeployNamespace)
+	}
+
+	serviceManifestPath := filepath.Join(df.RepoDir, olmManifestsDir, registryServiceManifestName)
+	addr, err := CreateRegistryService(df.Logger, df.Client, serviceManifestPath, registryDeployNamespace)
+	if err != nil {
+		return "", "", fmt.Errorf("failed to create the registry service manifest in the %s namespace: %v", err, registryDeployNamespace)
+	}
+	if addr == "" {
+		return "", "", fmt.Errorf("the registry service spec.ClusterIP returned is empty")
+	}
```

Comment:
We should probably reverse the creation of resources here as it's traditionally good practice to instantiate the service object before the deployment/pod that service is responsible for routing traffic.

Reply:
Done